### PR TITLE
Make message_events build for ros2 (on windows 10)

### PR DESCRIPTION
**Pull request by bfjelds**
Should go to a ROS2 branch.

Working towards enabling laser_filters in ROS2. This provides message_events for message_filters.

### CODE REVIEW

**Comment by dirk-thomas** [comment]
Thank you for starting this PR.

I am not really sure I understand the scope of the patch though. The packages in this repo are very specific to the generated C++ code of ROS 1 messages. Even with the proposed changes those wouldn't be used in ROS 2 messages anywhere. So I currently don't see how this would be useful.

Maybe you can provide a more detailed context, describe the goal as well as provide steps how to reproduce what you are trying to achieve with this (and related) PRs.

**Comment by bfjelds** [comment]
My end goal was really to get laser_filters working, so I generally just chipped away at what needed porting without too much thought towards refactoring or bigger picture.  This was created because rostime, message_event and the traits were needed to enable laser_filters.

Note there are a couple other PRs included in the work to get laser_filters going:

https://github.com/ros/filters/pull/19
https://github.com/ros2/geometry2/pull/53/

File: cpp_common/src/header.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@
  *   * Neither the name of Willow Garage, Inc. nor the names of its
  *     contributors may be used to endorse or promote products derived
  *     from this software without specific prior written permission.
- *
+ *90
```
Comment (wjwwood) [inline cpp_common/src/header.cpp:20]: typo?

Comment (bfjelds) [reply cpp_common/src/header.cpp:20]: definitely :)

**Comment by dirk-thomas** [comment]
Please check if the upcoming changes in ros2/message_filters#2 are covering all the cases of this patch.

**Comment by dirk-thomas** [comment]
Closing in favor of ros2/message_filters#5.

**Pull request closed without merging**